local/gugcat/tests/gugcat_test.php

```diff
@@ -262,4 +262,78 @@ public function test_blind_marking() {
         // False for manager role regardless of course module
         $this->assertFalse(local_gugcat::is_blind_marking($this->cm));
     }
+
+    public function test_switch_display_of_assessments_on_student_dashboard_without_customfield_category(){
+        global $DB;
+
+        $contextid = $this->coursecontext->id;
+        $instanceid = $this->course->id;
+
+        $switchdisplay = local_gugcat::switch_display_of_assessment_on_student_dashboard($instanceid, $contextid);
+
+        $this->assertEquals(1, $switchdisplay);
+    }
+
+    public function test_switch_display_of_assessments_on_student_dashboard_with_customfield_category_and_customfield_data(){
+        global $DB;
+
+        $contextid = $this->coursecontext->id;
+        $instanceid = $this->course->id;
+
+        $customfieldcategoryobj = self::default_custom_field_category_object();
+        $customfieldcategoryid = $DB->insert_record('customfield_category', $customfieldcategoryobj);
+
+        self::create_custom_field_field($customfieldcategoryid);
+
+        $customfieldfield = $DB->get_record('customfield_field', array('categoryid' => $customfieldcategoryid));
+
+        $customfieldddata = local_gugcat::default_contextfield_data_value($customfieldfield->id, $instanceid, $contextid);
+        $customfielddata = $DB->insert_record('customfield_data', $customfieldddata);
+
+        $switchdisplay = local_gugcat::switch_display_of_assessment_on_student_dashboard($instanceid, $contextid);
+
+        $this->assertEquals(0, $switchdisplay);
```

Comment:
Please add another assert here to test switch_display_of_assessment_on_student_dashboard will really toggle the assessment display (0/1)